src/protocol.c

```diff
@@ -136,8 +138,16 @@ thread_run_command(void *args) {
                 perror("setenv");
                 exit(1);
             }
-            if (execvp(server->argv[0], server->argv) < 0) {
-                perror("execvp");
+            int e = -1;
+            if ( 0 == access(server->argv[0], R_OK | X_OK) ) {
```

Comment:
imho it s bad to reinvent the wheel, system libc code looks better source to me that yafl ( yet another fancy lib )
More with less is a great motto for code.

https://github.com/openbsd/src/blob/master/lib/libc/stdlib/system.c#L60

What gain hte program will make by cutting the command line ? if you want to sake a fork you can put /bin/bash and teh system will not call a shell
But with this you can directly put shell command line like while true; !